WcaOnRails/spec/controllers/delegate_reports_controller_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+describe DelegateReportsController do
+  let(:comp) { FactoryGirl.create(:competition, :with_delegate) }
+
+  context "not logged in" do
+    it "redirects to sign in" do
+      get :show, competition_id: comp.id
+      expect(response).to redirect_to(new_user_session_path)
+    end
+  end
+
+  context "logged in as a regular user" do
+    sign_in { FactoryGirl.create(:user) }
+
+    it "redirects to home page" do
+      get :show, competition_id: comp.id
+      expect(response).to redirect_to(root_url)
+    end
+  end
+
+  context "logged in as a regular delegate" do
+    sign_in { FactoryGirl.create(:delegate) }
+
+    it "redirects to home page" do
+      get :edit, competition_id: comp.id
+      expect(response).to redirect_to(root_url)
+    end
+
+    it "redirects to home page" do
+      patch :update, competition_id: comp.id
+      expect(response).to redirect_to(root_url)
+    end
+  end
+
+  context "logged in as THE delegate" do
+    before :each do
+      sign_in comp.delegates.first
+    end
+
+    it "can view edit page" do
+      get :edit, competition_id: comp.id
+      expect(response.status).to eq 200
+    end
+
+    it "can edit report" do
+      post :update, competition_id: comp.id, delegate_report: { content: "My new content" }
+      expect(response).to redirect_to(competition_report_edit_path(comp))
+      comp.reload
+      expect(comp.delegate_report.content).to eq "My new content"
```

Comment:
Yeah, I prefer the explicit call to reload first. It keeps the subsequent lines shorter, and you don't have to worry about calling reload multiple times, or making sure that the line that *does* have a call to reload comes first.